lib/storage/file.js

```diff
@@ -980,6 +1002,50 @@ File.prototype.download = function(options, callback) {
 };
 
 /**
+ * The Storage API allows you to use a custom key for server-side encryption.
+ * Supply this method with a passphrase and the correct key (AES-256) will be
+ * generated and used for you.
+ *
+ * @resource [Customer-supplied Encryption Keys]{@link https://cloud.google.com/storage/docs/encryption#customer-supplied}
+ *
+ * @param {string} code - A passphrase to create a compliant key.
+ * @return {module:storage/file}
+ *
+ * @example
+ * var passphrase = 'encryption-code';
+ *
+ * var encryptedFile = myBucket.file('my-file').encrypt(passphrase);
+ * var unencryptedFile = myBucket.file('my-file');
+ *
+ * encryptedFile.save('data', function(err) {
+ *   unencryptedFile.download(function(err) {
+ *     // err: Bad Request
+ *
+ *     // Try again with the encrypted file:
+ *     encryptedFile.download(function(err, contents) {
+ *       // contents.toString() === 'data'
+ *     });
+ *   });
+ * });
+ */
+File.prototype.encrypt = function(code) {
```

Comment:
@stephenplusplus I'm worried about calling this "encrypt" and hinting to people that we're encrypting this before sending it to GCS, which we aren't. We're sending the file in plaintext, with an instruction to Google saying "please encrypt with this key for me...".

Can we make this a "setting" of sorts rather than the verb "encrypt" ?